### PR TITLE
PWA service workers

### PR DESCRIPTION
**Pull request by gparant**
- [x] Register service workers of PWA to install WorkAdventure application on desktop and mobile
- [x] Create specific web page for PWA
- [x] Add register service to save and redirect on a last card visited

### CODE REVIEW

File: front/src/Connexion/ConnectionManager.ts [hunk 1]
```diff
@@ -51,7 +56,7 @@ class ConnectionManager {
                 )
             );
             urlManager.pushRoomIdToUrl(room);
-            return Promise.resolve(room);
+            Promise.resolve(room);
```
Comment (Kharhamel) [inline front/src/Connexion/ConnectionManager.ts:59]: this line does nothing

File: front/src/Connexion/ConnectionManager.ts [hunk 2]
```diff
@@ -107,10 +112,14 @@ class ConnectionManager {
...
             }
-            return Promise.resolve(room);
+        }
+        if (room == undefined) {
+            return Promise.reject(new Error("Invalid URL"));
         }
 
-        return Promise.reject(new Error("Invalid URL"));
+        console.log("initGameConnexion => serviceWorker", room);
+        this.serviceWorker = new _ServiceWorker();
```
Comment (Kharhamel) [inline front/src/Connexion/ConnectionManager.ts:121]: are you sure you need to create the service worker here? Service workers are not supposed to interact directly with the rest of the codebase. Maybe create it in a separate file?

File: front/src/Connexion/ConnectionManager.ts [hunk 3]
```diff
@@ -107,10 +112,14 @@ class ConnectionManager {
...
                 localUserStore.saveUser(localUser);
             }
-            return Promise.resolve(room);
+        }
+        if (room == undefined) {
+            return Promise.reject(new Error("Invalid URL"));
         }
 
-        return Promise.reject(new Error("Invalid URL"));
+        console.log("initGameConnexion => serviceWorker", room);
```
Comment (Kharhamel) [inline front/src/Connexion/ConnectionManager.ts:120]: another console.log :(

**Review by Kharhamel — CHANGES_REQUESTED** [review]

File: front/src/Connexion/ConnectionManager.ts (hunk 1 above)
Comment (gparant) [reply front/src/Connexion/ConnectionManager.ts:59]: The function `initGameConnexion` is `initGameConnexion(): Promise<Room>`

**Pull request merged**